RadixWallet/Clients/SecurityCenterClient/SecurityCenterClient+Live.swift

```diff
@@ -0,0 +1,93 @@
+import AsyncAlgorithms
+import CloudKit
+import ComposableArchitecture
+import DependenciesAdditions
+import os
+
+extension SecurityCenterClient {
+	public static let liveValue: Self = .live()
+
+	public static func live(
+		profileStore: ProfileStore = .shared
+	) -> SecurityCenterClient {
+		@Dependency(\.userDefaults) var userDefaults
+		@Dependency(\.deviceFactorSourceClient) var deviceFactorSourceClient
+
+		@Sendable
+		func manualBackups() async -> AnyAsyncSequence<BackupStatus> {
+			let profileID = await profileStore.profile.id
+			let backups = userDefaults.lastManualBackupValues(for: profileID)
+			return await statusValues(results: backups)
+		}
+
+		@Sendable
+		func cloudBackups() async -> AnyAsyncSequence<BackupStatus> {
+			let profileID = await profileStore.profile.id
+			let backups = userDefaults.lastCloudBackupValues(for: profileID)
+			return await statusValues(results: backups)
+		}
+
+		@Sendable
+		func statusValues(results: AnyAsyncSequence<BackupResult>) async -> AnyAsyncSequence<BackupStatus> {
+			await combineLatest(profileStore.values(), results).map { profile, backup in
+				let upToDate = backup.profileHash == profile.hashValue
+				let success = backup.result == .success
+				return .init(backupDate: backup.backupDate, upToDate: upToDate, success: success)
+			}
+			.eraseToAnyAsyncSequence()
+		}
+
+		return .init(
+			problems: {
```

Comment:
The logic here too ought to done in Sargon IMO.

Reply:
I am not sure I see the benefit of that, this uses iOS specific APIs like user defaults and the logic is specific to iCloud backups. Also, can uniffi expose AsyncSequences?